pkg/operator/reporting/templates.go

```diff
@@ -42,7 +43,12 @@ func (ctx *ReportQueryTemplateContext) dataSourceTableName(name string) (string,
 			}
 			for _, prestoTable := range ctx.PrestoTables {
 				if prestoTable.Name == ds.Status.TableRef.Name {
-					return reportingutil.FullyQualifiedTableName(prestoTable), nil
+					dsTableName := reportingutil.FullyQualifiedTableName(prestoTable)
+
+					if strings.HasPrefix(dsTableName, ".") || strings.HasSuffix(dsTableName, ".") || strings.Contains(dsTableName, "..") {
```

Comment:
You don't need to do string matching, you can just return an error from `FullyQualifiedTableName` instead and return that here.

Reply:
Fixed.